src/safeposix/syscalls/net_calls.rs

```diff
@@ -1034,162 +1232,216 @@ impl Cage {
         }
     }
 
-    //TODO: handle pipes
     pub fn select_syscall(&self, nfds: i32, readfds: &mut interface::RustHashSet<i32>, writefds: &mut interface::RustHashSet<i32>, exceptfds: &mut interface::RustHashSet<i32>, timeout: Option<interface::RustDuration>) -> i32 {
-        //exceptfds and writefds are not really implemented at the current moment.
-        //They both always return success. However we have some intention of making
-        //writefds work at some point for pipes? We have no such intention for exceptfds
-        let new_readfds = interface::RustHashSet::<i32>::new();
-        let new_writefds = interface::RustHashSet::<i32>::new();
-        //let mut new_exceptfds = interface::RustHashSet::<i32>::new(); we don't ever support exceptional conditions
-    
-        if nfds < STARTINGFD || nfds >= MAXFD {
-            return syscall_error(Errno::EINVAL, "select", "Number of FDs is wrong");
-        }
-    
-        let start_time = interface::starttimer();
-    
-        let end_time = match timeout {
-            Some(time) => time,
-            None => interface::RustDuration::MAX
-        };
+       //exceptfds and writefds are not really implemented at the current moment.
+       //They both always return success. However we have some intention of making
+       //writefds work at some point for pipes? We have no such intention for exceptfds
+       let mut new_readfds = interface::RustHashSet::<i32>::new();
+       let mut new_writefds = interface::RustHashSet::<i32>::new();
+       //let mut new_exceptfds = interface::RustHashSet::<i32>::new(); we don't ever support exceptional conditions
+   
+       if nfds < STARTINGFD || nfds >= MAXFD {
+           return syscall_error(Errno::EINVAL, "select", "Number of FDs is wrong");
+       }
+   
+       let start_time = interface::starttimer();
+   
+       let end_time = match timeout {
+           Some(time) => time,
+           None => interface::RustDuration::MAX
+       };
  
-        let mut retval = 0; 
-        loop { //we must block manually
-            for fd in readfds.iter() {
+       let mut retval = 0; 
+       loop { //we must block manually
+           
+            // 1. iterate thru readfds
+            let mut res = self.select_readfds(readfds, &mut new_readfds, &mut retval);
```

Comment:
Just curious, I forget why this change with select was needed. Was it due to read/write locks causing issues?

Reply:
Are you asking why we refactored to separate the read and write parts of select? Mostly for readability.